### PR TITLE
Added the markdown in channel boardSelector RHS for description

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
#### Summary
This PR adds the markdown Editor to the channel BoardSelector RHS for description.

#### Ticket Link
Fixes https://github.com/mattermost/focalboard/issues/3661

### CODE REVIEW

**Comment by sbishel** [comment]
@Rajat-Dabade - If this is ready for review, please add reviewer to this PR.

**Review by Pinjasaur — CHANGES_REQUESTED** [review]
@Rajat-Dabade overall looks good, just a few small items

1. mentioned updating a test to assert markdown syntax rendering
2. looks like a snapshot needs updating
3. can you update the styling so the cursor is a pointer even when hovering the `<MarkdownEditor>`? see screenshot

image

**Comment by sbishel** [comment]
@Rajat-Dabade - Lets get this finished up and merged.

**Comment by Pinjasaur** [comment]
@Rajat-Dabade a couple questions

1. my memory is fuzzy, but I thought we had an all-team or dev meeting pre-Mattercon where we talked about this PR and (I _think_ @jespino?) discussed that using e.g. `Utils.htmlFromMarkdown` would be more appropriate since we don't need a heavyweight `<MarkdownEditor>` that's in `readonly` mode
2. if my previous statement is incorrect, we should still be able to do snapshot testing by mocking that `data-offset-key` with a deterministic value e.g. https://github.com/facebook/draft-js/issues/702#issuecomment-283169324

**Comment by Rajat-Dabade** [comment]
Ah, I almost forgot @Pinjasaur, Thanks for reminding me, I have updated the code, please take a look.

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.scss [hunk 1]
```diff
@@ -38,5 +38,10 @@
         -webkit-line-clamp: 3;
         -webkit-box-orient: vertical;
         overflow: hidden;
+        cursor: pointer;
+
+        .MarkdownEditor {
```
Comment (sbishel) [inline mattermost-plugin/webapp/src/components/rhsChannelBoardItem.scss:43]: I believe this object `.MarkdownEditor` was removed...so this is no longer necessary.

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.tsx [hunk 2]
```diff
@@ -47,6 +47,7 @@ const RHSChannelBoardItem = (props: Props) => {
 
     const untitledBoardTitle = intl.formatMessage({id: 'ViewTitle.untitled-board', defaultMessage: 'Untitled board'})
 
+    const html = Utils.htmlFromMarkdown(board.description)
```
Comment (sbishel) [inline mattermost-plugin/webapp/src/components/rhsChannelBoardItem.tsx:50]: nit: I don't like the variable name `html`. Change to something more descriptive. ie. `markdownHtml`

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx [hunk 3]
```diff
@@ -73,6 +85,9 @@ describe('components/rhsChannelBoardItem', () => {
         const buttonElement = screen.getByRole('button', {name: 'menuwrapper'})
         await userEvent.click(buttonElement)
 
+        const description = getByText(/with description/i)
+
+        expect(description.outerHTML).toEqual("<p><strong>Board</strong> with description</p>")
```
Comment (sbishel) [inline mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx:90]: This check is not necessary as the snapshot check below, should be doing exactly that.

File: mattermost-plugin/webapp/src/components/__snapshots__/rhsChannelBoardItem.test.tsx.snap [hunk 4]
```diff
@@ -77,10 +81,11 @@ exports[`components/rhsChannelBoardItem render board with menu open 1`] = `
             <div
               class="menu-options"
             >
+              <div />
               <div>
                 <div
-                  aria-label="Unlink board"
-                  class="MenuOption TextOption menu-option"
+                  aria-label="Unlink board Hello"
+                  class="MenuOption TextOption menu-option menu-option--with-subtext menu-option--disabled"
```
Comment (sbishel) [inline mattermost-plugin/webapp/src/components/__snapshots__/rhsChannelBoardItem.test.tsx.snap:88]: Why did this change?

File: mattermost-plugin/webapp/src/components/__snapshots__/rhsChannelBoardItem.test.tsx.snap [hunk 5]
```diff
@@ -100,15 +105,19 @@ exports[`components/rhsChannelBoardItem render board with menu open 1`] = `
                     <div
                       class="menu-name"
                     >
-                      Unlink board
+                      Unlink board Hello
+                    </div>
+                    <div
+                      class="menu-subtext text-75 mt-1"
+                    >
+                      You are not an admin of the board
```
Comment (sbishel) [inline mattermost-plugin/webapp/src/components/__snapshots__/rhsChannelBoardItem.test.tsx.snap:113]: Why did this change?

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx [hunk 6]
```diff
@@ -52,19 +54,29 @@ describe('components/rhsChannelBoardItem', () => {
...
-            }
+            },
+            users: {
+                boardUsers: {
+                    user_id_1: {username: 'abc'},
+                },
+            },
+            clientConfig: {
+                value: {},
+            },
```
Comment (sbishel) [inline mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx:72]: Same here, are these changes necessary? Or only necessary because of changes to "TestBlockFactory"?

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx [hunk 7]
```diff
@@ -42,7 +44,7 @@ describe('components/rhsChannelBoardItem', () => {
     })
 
     it('render board with menu open', async () => {
-        const board = createBoard()
+        const board = TestBlockFactory.createBoard()
```
Comment (sbishel) [inline mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx:47]: Guessing this was the reason for the snapshot changes above.

What was the reason behind changing to TestBlockFactory? If it was necessary for proper testing, then I would also change the other place we currently call `createBoard()`.
If not necessary, I would leave, as it causes unnecessary changes to snapshots which confuse the actually changes to this PR.

**Review by sbishel — CHANGES_REQUESTED** [review]
I would suggest minimizing the changes for this. It seems that snapshots have changed unnecessarily.

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx (hunk 6 above)
Comment (Rajat-Dabade) [reply mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx:72]: These changes were required for the MarkdownEditor component. Forgot to remove it. Will remove it. Thanks.

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx (hunk 3 above)
Comment (Rajat-Dabade) [reply mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx:90]: Ah, got it. Will remove it. Thanks.

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.tsx (hunk 2 above)
Comment (Rajat-Dabade) [reply mattermost-plugin/webapp/src/components/rhsChannelBoardItem.tsx:50]: Make sense, will change it.

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.scss (hunk 1 above)
Comment (Rajat-Dabade) [reply mattermost-plugin/webapp/src/components/rhsChannelBoardItem.scss:43]: Okay.

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx (hunk 7 above)
Comment (sbishel) [reply mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx:45]: That's great. Let's go ahead and update the other test to `TestBlockFactory.createBoard()` as well and eventually we can deprecate it...also may be a good community campaign to change it in other tests.

**Review by sbishel — APPROVED** [review]
You can move the test updates to separate PR if desired.

File: mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx [hunk 8]
```diff
@@ -43,6 +43,7 @@ describe('components/rhsChannelBoardItem', () => {
 
     it('render board with menu open', async () => {
         const board = createBoard()
+        board.description = 'New board'
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply mattermost-plugin/webapp/src/components/rhsChannelBoardItem.test.tsx]: Done.

**Comment by Rajat-Dabade** [comment]
Sure 👍

**Review by Pinjasaur — APPROVED** [review]
LGTM, thanks @Rajat-Dabade ✨
